internal/stacksapi/client_test.go

```diff
@@ -0,0 +1,198 @@
+package stacksapi
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/buildkite/agent-stack-k8s/v2/internal/version"
+	"github.com/buildkite/roko"
+)
+
+func TestNewClient(t *testing.T) {
+	t.Parallel()
+	t.Run("creates client with defaults", func(t *testing.T) {
+		t.Parallel()
+		client, err := NewClient("test-token")
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		if client.baseURL.String() != DefaultBaseURL {
+			t.Errorf("expected baseURL %s, got %s", DefaultBaseURL, client.baseURL)
+		}
+
+		expectedUA := "agent-stack-k8s/" + version.Version()
+		if client.userAgent != expectedUA {
+			t.Errorf("expected userAgent %s, got %s", expectedUA, client.userAgent)
+		}
+
+		expectedAuth := "Token test-token"
+		if client.authHeader != expectedAuth {
+			t.Errorf("expected authHeader %s, got %s", expectedAuth, client.authHeader)
+		}
+	})
+
+	t.Run("applies client options", func(t *testing.T) {
+		t.Parallel()
+		customURL := urlMustParse("https://api.example.com/")
+		customClient := &http.Client{Timeout: 10 * time.Second}
+		logger := slog.New(slog.NewTextHandler(nil, nil))
+
+		client, err := NewClient("test-token",
+			WithBaseURL(customURL),
+			WithHTTPClient(customClient),
+			WithLogger(logger),
+			PrependToUserAgent("custom-agent"),
+		)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		if client.baseURL.String() != customURL.String() {
+			t.Errorf("expected baseURL %s, got %s", customURL, client.baseURL)
+		}
+
+		if client.httpClient != customClient {
+			t.Error("expected custom HTTP client to be set")
+		}
+
+		expectedUA := "custom-agent agent-stack-k8s/" + version.Version()
+		if client.userAgent != expectedUA {
+			t.Errorf("expected userAgent %s, got %s", expectedUA, client.userAgent)
+		}
+	})
+}
+
+func TestNewRequest(t *testing.T) {
+	t.Parallel()
+	client, err := NewClient("test-token")
+	if err != nil {
+		t.Fatalf("failed to create client: %v", err)
+	}
+
+	t.Run("creates request with proper headers", func(t *testing.T) {
+		t.Parallel()
+		ctx := context.Background()
+		req, err := client.newRequest(ctx, "GET", "test", nil)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		if req.Method != "GET" {
+			t.Errorf("expected method GET, got %s", req.Method)
+		}
+
+		expectedURL := "https://agent.buildkite.com/v3/test"
+		if req.URL.String() != expectedURL {
+			t.Errorf("expected URL %s, got %s", expectedURL, req.URL.String())
+		}
+
+		if req.Header.Get("Authorization") != "Token test-token" {
+			t.Error("Authorization header not set correctly")
+		}
```

Comment:
Not set correctly, so what was it set to?

```suggestion
		if got, want := req.Header.Get("Authorization"), "Token test-token" {
			t.Error("req.Header.Get(\"Authorization\") = %q, want %q", got, want)
		}
```